src/channels.ml

```diff
@@ -96,11 +96,24 @@ let of_seekable_fd fd =
     return () in
   return { c with skip }
 
-let sslctx =
-  Ssl.init ();
-  Ssl.create_context Ssl.SSLv23 Ssl.Client_context
+let _ =
+  Ssl.init ()
 
-let of_ssl_fd fd =
+let legacy_sslctx good_ciphersuites legacy_ciphersuites =
+  let ctx = Ssl.create_context Ssl.SSLv23 Ssl.Client_context in
+  Ssl.set_cipher_list ctx (good_ciphersuites ^ (match legacy_ciphersuites with "" -> "" | s -> (":" ^ s)));
+  Ssl.disable_protocols ctx [Ssl.SSLv3];
+  ctx
+
+let good_sslctx good_ciphersuites =
+  let ctx = Ssl.create_context Ssl.TLSv1_2 Ssl.Client_context in
+  Ssl.set_cipher_list ctx good_ciphersuites;
+  ctx
+
+let of_ssl_fd fd ssl_legacy good_ciphersuites legacy_ciphersuites =
+  let good_ciphersuites = match good_ciphersuites with None -> failwith "good_ciphersuites not specified" | Some x -> x in
+  let legacy_ciphersuites = match legacy_ciphersuites with None -> failwith "legacy_ciphersuites not specified" | Some x -> x in
```

Comment:
Specifying legacy_ciphersuites should be optional. So this line should be
`let legacy_ciphersuites = match legacy_ciphersuites with None -> "" | Some x -> x in`